pkg/exporter.go

```diff
@@ -155,7 +156,8 @@ func UpdateMetrics(
 
 	metrics = append(metrics, promutil.BuildNamespaceInfoMetrics(tagsData, options.labelsSnakeCase, logger)...)
 
-	registry.MustRegister(promutil.NewPrometheusCollector(metrics))
+	collector := promutil.NewPrometheusCollector(metrics)
+	registry.MustRegister(collector)
```

Comment:
Given the complexity of the duplicate metric code I was surprised it didn't show up in pprof. After I made this change it started appearing. I have no idea why this is the case though.